src/libs/Navigation/AppNavigator/Navigators/OnboardingModalNavigator.tsx

```diff
@@ -0,0 +1,43 @@
+import {createStackNavigator} from '@react-navigation/stack';
+import React from 'react';
+import {View} from 'react-native';
+import NoDropZone from '@components/DragAndDrop/NoDropZone';
+import useOnboardingLayout from '@hooks/useOnboardingLayout';
+import useThemeStyles from '@hooks/useThemeStyles';
+import OnboardingModalNavigatorScreenOptions from '@libs/Navigation/AppNavigator/OnboardingModalNavigatorScreenOptions';
+import type {OnboardingModalNavigatorParamList} from '@libs/Navigation/types';
+import OnboardingPersonalDetails from '@pages/OnboardingPersonalDetails';
+import OnboardingPurpose from '@pages/OnboardingPurpose';
+import SCREENS from '@src/SCREENS';
+import Overlay from './Overlay';
+
+const Stack = createStackNavigator<OnboardingModalNavigatorParamList>();
+
+function OnboardingModalNavigator() {
+    const styles = useThemeStyles();
+    const {shouldUseNarrowLayout} = useOnboardingLayout();
+
+    return (
+        <NoDropZone>
+            <Overlay />
+            <View style={styles.onboardingNavigatorOuterView}>
+                <View style={styles.OnboardingNavigatorInnerView(shouldUseNarrowLayout)}>
+                    <Stack.Navigator screenOptions={OnboardingModalNavigatorScreenOptions()}>
+                        <Stack.Screen
+                            name={SCREENS.ONBOARDING.PERSONAL_DETAILS}
+                            component={OnboardingPersonalDetails}
+                        />
+                        <Stack.Screen
+                            name={SCREENS.ONBOARDING.PURPOSE}
+                            component={OnboardingPurpose}
+                        />
```

Comment:
I haven't tried that yet, I'll see whether it fixes the problem.

Reply:
It does fix the typecheck errors and we should definitely use this for now, however this does seem more like a workaround than a fix, since it uses a generalized type to make TS ignore any prop-specific errors in the navigator:

`type OnboardingPurposeProps = Record<string, unknown>;`